packages/dosespot-react/src/DoseSpotNewMedicationForm.tsx

```diff
@@ -0,0 +1,154 @@
+import { Box, Button, Divider, Group, NumberInput, Stack, Text, TextInput } from '@mantine/core';
+import { NDC, RXNORM } from '@medplum/core';
+import { Bundle, MedicationKnowledge } from '@medplum/fhirtypes';
+import React, { useState } from 'react';
+import { DOSESPOT_DISPENSABLE_DRUG_ID_SYSTEM, DOSESPOT_REFILLS_SYSTEM } from './common';
+import { DoseSpotMedicationSelect } from './DoseSpotMedicationSelect';
+
+export interface DoseSpotNewMedicationFormProps {
+  readonly searchMedications: (term: string) => Promise<Bundle<MedicationKnowledge> | undefined>;
+  readonly addFavoriteMedication: (medication: MedicationKnowledge) => Promise<void>;
+  readonly loading?: boolean;
+}
+
+interface FormData {
+  quantity: number;
+  refills: number;
+  directions: string;
+}
+
+export function DoseSpotNewMedicationForm(props: DoseSpotNewMedicationFormProps): React.JSX.Element {
+  const { searchMedications, addFavoriteMedication, loading = false } = props;
+  const [formData, setFormData] = useState<FormData>({
+    quantity: 1,
+    refills: 0,
+    directions: '',
+  });
+  const [medication, setMedication] = useState<MedicationKnowledge | null>(null);
+
+  const handleAddFavorite = async (): Promise<void> => {
```

Comment:
@techdavidy could you comment on this - something feels off to me for a React callback to both depend on `medication` and to call `setMedication`, but I'm not a React expert. 

At the very least, seems like `useCallback` should be used uer